internal/service/federatedsettingsidentityprovider/resource_federated_settings_identity_provider.go

```diff
@@ -112,15 +112,14 @@ func Resource() *schema.Resource {
 	}
 }
 
+func noOpCreate(_ context.Context, _ *schema.ResourceData, _ any) diag.Diagnostics {
+	return diag.FromErr(errors.New("this resource must be imported"))
```

Comment:
knit: not sure "no-op" is a good name as it really does something, throws an error, maybe something like resourceCreateNotAllowed.

you might also want to change resourceMongoDBAtlasFederatedSettingsIdentityProviderRead to resourceRead, etc.

Reply:
Agree :D Changed in [b4964e6](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2290/commits/b4964e64fcf3b7e5694bb425719387c29179e7b4)
Saving the other method naming to next weeks epic 👍